ProcessSmarterTestPackage/PostProcessors/FormPartitionPostProcessor.cs

```diff
@@ -16,23 +16,101 @@ public override IList<ValidationError> GenerateErrors()
                         GeneratedMessage =
-                            $"[Itemgroup identifier {itemGroup.ChildNodeWithName("identifier").ValueForAttribute("uniqueid")} does not follow expected pattern {partitionIdentifier}:G-<ID>]",
+                            $"[Itemgroup identifier {itemGroup.ChildNodeWithName("identifier").ValueForAttribute("uniqueid")} does not follow expected pattern {groupIdPattern}]",
                         Key = "uniqueid",
                         ItemId = itemGroup.ChildNodeWithName("identifier").ValueForAttribute("uniqueid"),
                         PackageType = PackageType,
                         Location = "itemgroup/identifier"
                     });
                 }
+                else
+                {
+                    if (!match.Groups[1].Value.Equals(partitionIdentifier))
+                    {
+                        result.Add(new ValidationError
+                        {
+                            ErrorSeverity = ErrorSeverity.Benign,
+                            GeneratedMessage =
+                                $"[Itemgroup identifier {itemGroup.ChildNodeWithName("identifier").ValueForAttribute("uniqueid")} first portion {match.Groups[1]} does not match partition ID and expected value {partitionIdentifier}]",
+                            Key = "uniqueid",
+                            ItemId = itemGroup.ChildNodeWithName("identifier").ValueForAttribute("uniqueid"),
+                            PackageType = PackageType,
+                            Location = "itemgroup/identifier"
+                        });
+                    }
+                    if (match.Groups[2].Value.Equals("G"))
+                    {
+                        if (itemGroup.ChildNodesWithName("passageref").Count() != 1)
+                        {
+                            result.Add(new ValidationError
+                            {
+                                ErrorSeverity = ErrorSeverity.Severe,
+                                GeneratedMessage =
+                                    $"[Itemgroup identifier {itemGroup.ChildNodeWithName("identifier").ValueForAttribute("uniqueid")} is a multi-item specified \"G-\" item group, but does not have a reference to a single child passageref element. This assessment will fail in TDS at runtime]",
+                                Key = "uniqueid",
+                                ItemId = itemGroup.ChildNodeWithName("identifier").ValueForAttribute("uniqueid"),
+                                PackageType = PackageType,
+                                Location = "itemgroup/identifier"
+                            });
+                        }
+                        else if (!match.Groups[3].Value.Equals(
+                            itemGroup.ChildNodeWithName("passageref").ValidatedAttributes["passageref"].Value))
+                        {
+                            result.Add(new ValidationError
```

Comment:
"G-" item group identifiers second portion should match their associated passage ID by convention.